app/src/main/java/org/schabi/newpipe/player/mediabrowser/MediaBrowserConnector.java

```diff
@@ -0,0 +1,328 @@
+        builder.setExtras(extras);
+        return new MediaItem(builder.build(), MediaItem.FLAG_BROWSABLE);
+    }
+
+    private MediaItem createPlaylistMediaItem(final PlaylistMetadataEntry playlist) {
+        final var builder = new MediaDescriptionCompat.Builder();
+        builder.setMediaId(createMediaIdForPlaylist(playlist.uid))
+            .setTitle(playlist.name)
+            .setIconUri(Uri.parse(playlist.thumbnailUrl));
+
+        final var extras = new Bundle();
+        extras.putString(MediaConstants.DESCRIPTION_EXTRAS_KEY_CONTENT_STYLE_GROUP_TITLE,
+                playerService.getResources().getString(R.string.tab_bookmarks));
+        builder.setExtras(extras);
+        return new MediaItem(builder.build(), MediaItem.FLAG_BROWSABLE);
+    }
+
+    private String createMediaIdForPlaylist(final long playlistId) {
+        return ID_BOOKMARKS + '/' + playlistId;
+    }
+
+    private MediaItem createPlaylistStreamMediaItem(final long playlistId,
+                                                    final PlaylistStreamEntry item,
+                                                    final int index) {
+        final var builder = new MediaDescriptionCompat.Builder();
+        builder.setMediaId(createMediaIdForPlaylistIndex(playlistId, index))
+                .setTitle(item.getStreamEntity().getTitle())
+                .setIconUri(Uri.parse(item.getStreamEntity().getThumbnailUrl()));
+
+         return new MediaItem(builder.build(), MediaItem.FLAG_PLAYABLE);
+    }
+
+    private String createMediaIdForPlaylistIndex(final long playlistId, final int index) {
+        return createMediaIdForPlaylist(playlistId) + '/' + index;
+    }
+
+    @Nullable
+    public MediaBrowserServiceCompat.BrowserRoot onGetRoot(@NonNull final String clientPackageName,
+                                                           final int clientUid,
+                                                           @Nullable final Bundle rootHints) {
+        if (DEBUG) {
+            Log.d(TAG, String.format("MediaBrowserService.onGetRoot(%s, %s, %s)",
+                  clientPackageName, clientUid, rootHints));
+        }
+
+        return new MediaBrowserServiceCompat.BrowserRoot(ID_ROOT, null);
+    }
+
+    public Single<List<MediaItem>> onLoadChildren(@NonNull final String parentId) {
+        if (DEBUG) {
+            Log.d(TAG, String.format("MediaBrowserService.onLoadChildren(%s)", parentId));
+        }
+
+        final List<MediaItem> mediaItems = new ArrayList<>();
+        final var parentIdUri = Uri.parse(parentId);
+
+        if (parentId.equals(ID_ROOT)) {
+            mediaItems.add(
+                    createRootMediaItem(ID_BOOKMARKS,
+                            playerService.getResources().getString(R.string.tab_bookmarks)));
+            mediaItems.add(
+                    createRootMediaItem(ID_HISTORY,
+                            playerService.getResources().getString(R.string.action_history)));
```

Comment:
`NewPipe` allows the user to modify the tabs to be shown on the home screen. Can we add similar functionality for the Auto home screen as well. Alternatively, we could decide to reflect the home page tabs for consistency. Everything does not have to be in this CL, just make provision to keep this generic rather than hard-coded options.

@AudricV What do you think

Reply:
Yes, this makes sense. I wasn't sure what the best structure for Auto browser would be, so I just started with a couple of examples.

Reply:
That's a great idea, but what would happen in the case the user disabled a supported tab or all supported tabs?

Reply:
One comment about the root menu: the [Android Auto documentation](https://developer.android.com/training/cars/media?hl=en#root-menu-structure) says that in most cases, there's a limit of 3-4 tabs, and additional tabs won't be shown. I haven't tried this, though. It would make it difficult to use the same tab configuration that users have chosen for their home screen on the phone.